File: app/src/main/java/com/worldventures/dreamtrips/wallet/service/nxt/util/DetokenizedRecord.java
package com.worldventures.dreamtrips.wallet.service.nxt.util;

import android.support.annotation.NonNull;

import com.worldventures.dreamtrips.wallet.domain.entity.record.Record;
import com.worldventures.dreamtrips.wallet.service.nxt.model.MultiResponseBody;

import java.util.Collections;
import java.util.List;

public class DetokenizedRecord extends NxtRecordResponse {

   public static DetokenizedRecord from(@NonNull Record tokenizedRecord, @NonNull MultiResponseBody nxtResponses) {
      return from(tokenizedRecord, Collections.singletonList(nxtResponses));
   }

   public static DetokenizedRecord from(@NonNull Record tokenizedRecord, @NonNull List<MultiResponseBody> nxtResponses) {
      return new DetokenizedRecord(tokenizedRecord, nxtResponses);
   }

   private DetokenizedRecord(@NonNull Record tokenizedRecord, @NonNull List<MultiResponseBody> nxtResponses) {
      super(tokenizedRecord, nxtResponses);
   }

   @Override
   public Record getTokenizedRecord() {
      return record;
   }

   @Override
   public Record getDetokenizedRecord() {
      return NxtBankCardHelper.getDetokenizedRecord(this, refIdPrefix);
   }

}

File: wallet/src/main/java/com/worldventures/wallet/service/nxt/util/DetokenizedRecord.kt
package com.worldventures.wallet.service.nxt.util

import com.worldventures.wallet.domain.entity.record.Record
import com.worldventures.wallet.service.nxt.model.MultiResponseBody

class DetokenizedRecord private constructor(tokenizedRecord: Record, nxtResponses: List<MultiResponseBody>)
   : NxtRecordResponse(tokenizedRecord, nxtResponses) {

   override fun getTokenizedRecord(): Record {
      return record
   }

   override fun getDetokenizedRecord(): Record {
      return NxtBankCardHelper.getDetokenizedRecord(this, refIdPrefix)
   }

   companion object {

      fun from(tokenizedRecord: Record, nxtResponses: MultiResponseBody): DetokenizedRecord {
         return from(tokenizedRecord, listOf(nxtResponses))
      }

      fun from(tokenizedRecord: Record, nxtResponses: List<MultiResponseBody>): DetokenizedRecord {
         return DetokenizedRecord(tokenizedRecord, nxtResponses)
      }
   }

}
